selectors/views/channel_sidebar.ts

```diff
@@ -75,23 +83,32 @@ export const getChannelsInCategoryOrder = (() => {
         getCategoriesForCurrentTeam,
         getChannelsByCategoryForCurrentTeam,
         getCurrentChannelId,
-        (state: GlobalState) => getUnreadChannelIds(state),
-        (collapsedState, categories, channelsByCategory, currentChannelId, unreadChannelIds) => {
+        getUnreadChannelIdsSet,
+        shouldShowUnreadsCategory,
+        (collapsedState, categories, channelsByCategory, currentChannelId, unreadChannelIds, showUnreadsCategory) => {
             return categories.map((category) => {
                 const channels = channelsByCategory[category.id];
                 const isCollapsed = collapsedState[category.id];
 
-                if (isCollapsed) {
-                    const filter = (channel: Channel) => {
-                        const filterByUnread = (channelId: string) => channel.id === channelId;
-                        const isUnread = unreadChannelIds.some(filterByUnread);
+                return channels.filter((channel: Channel) => {
+                    const isUnread = unreadChannelIds.has(channel.id);
 
-                        return isUnread || currentChannelId === channel.id;
-                    };
-                    return channels.filter(filter);
-                }
+                    if (showUnreadsCategory) {
```

Comment:
I'm leaving these split out for readability since it's a bit easier to see this way that this works differently when the unreads category is visible. It's not strictly necessary here, but I find it more helpful below when the combined condition would be more complicated.